subsys/net/ip/net_if.c

```diff
@@ -3076,13 +3076,18 @@ static uint8_t get_diff_ipv6(const struct in6_addr *src,
 
 static inline bool is_proper_ipv6_address(struct net_if_addr *addr)
 {
-	if (addr->is_used && addr->addr_state == NET_ADDR_PREFERRED &&
-	    addr->address.family == AF_INET6 &&
+	struct net_if_addr *candidate = NULL;
+
+	if (addr->is_used && addr->address.family == AF_INET6 &&
 	    !net_ipv6_is_ll_addr(&addr->address.in6_addr)) {
-		return true;
+		if (addr->addr_state == NET_ADDR_PREFERRED) {
+			return true;
+		} else if (addr->addr_state == NET_ADDR_DEPRECATED) {
+			candidate = addr;
```

Comment:
We're not looping here, so wouldn't we get the same effect by simply returning `true` here as well, w/o introducing a variable?

But moreover, I'm not sure if that change is actually enough? At the `net_if_ipv6_get_best_match()` level, we won't differentiate between preferred and depracted addresses, I. e. a depracated address with a better prefix match would take precedence over a preferred address (which if I understand correctly shouldn't be the case)? I think `net_if_ipv6_get_best_match()` should take into account whether it deals with a deprecated address or not, and only consider using deprecated one if no preferred address was found so far.

Reply:
Hmm, good point. I will need to adjust the PR more, I will send a better version.